### ISSUE TITLE
False positive in Next.js Config file: `next.config.js`

### ISSUE THREAD

**Issue by deadcoder0904** [depcheck/depcheck#583]
I have the following `next.config.js`:

```js
const { createLoader } = require('simple-functional-loader')
const rehypePrism = require('@mapbox/rehype-prism')
const withBundleAnalyzer = require('@next/bundle-analyzer')({
	enabled: process.env.ANALYZE === 'true',
})

module.exports = withBundleAnalyzer({
	pageExtensions: ['js', 'jsx', 'ts', 'tsx', 'md', 'mdx'],
	experimental: {
		modern: true,
	},
	webpack: (config, options) => {
		config.module.rules.push({
			test: /\.(svg|png|jpe?g|gif|mp4)$/i,
			use: [
				{
					loader: 'file-loader',
					options: {
						publicPath: '/_next',
						name: 'static/media/[name].[hash].[ext]',
					},
				},
			],
		})

		const mdx = [
			options.defaultLoaders.babel,
			{
				loader: '@mdx-js/loader',
				options: {
					rehypePlugins: [rehypePrism],
				},
			},
		]

		config.module.rules.push({
			test: /\.mdx$/,
			oneOf: [
				{
					resourceQuery: /preview/,
					use: [
						...mdx,
						createLoader(function (src) {
							if (src.includes('<!--more-->')) {
								const [preview] = src.split('<!--more-->')
								return this.callback(null, preview)
							}

							const [preview] = src.split('<!--/excerpt-->')
							return this.callback(null, preview.replace('<!--excerpt-->', ''))
						}),
					],
				},
				{
					use: [
						...mdx,
						createLoader(function (src) {
							const content = [
								'import Blog from "@/components/Blog"',
								'export { getStaticProps } from "@/utils/blog/getStaticProps"',
								'export { getStaticPaths } from "@/utils/blog/getStaticPaths"',
								src,
								'export default (props) => <Blog meta={meta} {...props} />',
							].join('\n')
							
							if (content.includes('<!--more-->')) {
								return this.callback(null, content.split('<!--more-->').join('\n'))
							}

							return this.callback(null, content.replace(/<!--excerpt-->.*<!--\/excerpt-->/s, ''))
						}),
					],
				},
			],
		})

		return config
	},
})
```

It has `@mdx-js/loader` in it but when I use `depcheck` it gives me a false positive:

```bash
➜ depcheck 
Unused dependencies
* @mdx-js/runtime
* gray-matter
Unused devDependencies
* @mdx-js/loader
```

I think it shouldn't show `@mdx-js/loader` :)

**Comment by rumpl** [comment]
@deadcoder0904 thanks for reporting this, would it be possible to have a repo that reproduces this error? I never used next.js and am not sure I want to spend too much time setting up a repo test case.

Thanks

**Comment by deadcoder0904** [comment]
Sure, just clone https://github.com/tailwindlabs/blog.tailwindcss.com to check the error.

**Comment by rumpl** [comment]
Nice, thanks!

**Comment by rumpl** [comment]
Oookay so the problem is that we don't support `oneOf` in webpack configuration file

**Comment by rumpl** [comment]
@deadcoder0904 this is fixed in main, could you test it on your project? I tested it on tailwind's blog and my PR fixes it

**Comment by rumpl** [comment]
Just reopened, if it's fixed, feel free to close the issue :)

**Comment by deadcoder0904** [comment]
How do I test it? Can you release the new version so I can update it?

**Comment by rumpl** [comment]
Version 1.3.0 with this change will soon be available on npm

**Comment by deadcoder0904** [comment]
And it works fine 🎉
